src/server/game/Movement/Spline/MovementPacketBuilder.cpp

```diff
@@ -98,7 +98,7 @@ namespace Movement
 
     void WriteLinearPath(const Spline<int32>& spline, ByteBuffer& data)
     {
-        uint32 last_idx = spline.getPointCount() - 3;
+        uint32 last_idx = spline.getPointCount() - 2;
```

Comment:
The way that spline ends is used by the client to determine if the path is linear or not. Etc.